### DISCUSSION TITLE
select for update with r2dbc driver

### DISCUSSION THREAD

**Discussion by crionuke** [pgjdbc/r2dbc-postgresql#442] [General]
Hi, there!
Need help, can i use pessimistic write locking via select for update statement with this r2dbc driver?

**Comment by mp911de** [comment]
Care to elaborate how that's specific to R2DBC? Isn't it a pure matter of the SQL that you send to the database?

**Comment by crionuke** [comment]
@mp911de 
just because found mentioning only for optimistic locking from spring docs - https://docs.spring.io/spring-data/r2dbc/docs/current/reference/html/#r2dbc.optimistic-locking
so could be any pitfalls with select for update locking mechanism and non blocking driver ?

**Reply by mp911de** [reply]
Thanks for the context. Pessimistic locking happens within the database which means that the driver doesn't receive a response until either the lock is released or the lock timeout is reached. In any case, the connection remains blocked and therefore not usable until lock release. That however doesn't affect any threads since the driver doesn't block threads. Pessimistic locking therefore can lead to increased latency.

**Reply by crionuke** [reply]
Thanks a lot for explanation!

**Comment by crionuke** [comment]
@mp911de 
Sorry, again.
There is any plans to implement some locking api for r2dbc (may be for example for R2dbcEntityTemplate)
as it was in classic spring data jpa like this
```
@Lock(LockModeType.PESSIMISTIC_WRITE)
List<User> findByLastname(String lastname);
```

**Comment by mp911de** [comment]
> There is any plans to implement some locking api for r2dbc (may be for example for R2dbcEntityTemplate)

That's a Spring topic, not related to the driver at all.

**Reply by crionuke** [reply]
sorry thought that you could answer here to reduce topics counts and save context of all thread.
